### ISSUE TITLE
@Comment cause ParserException::unexpectedCharacter

### ISSUE THREAD

**Issue by Nahihilo** [renanbr/bibtex-parser#49]
Hi,

I'm working on a .bib file that contain some "@Comment{}" and when the parser arrive to one of them, it cause an error, because they don't have to be valid entry and some aren't...

> Actually, the rule is that everything from the @Comment and to the end of line is ignored. The remainder lines of the commented entry is ignored by the first comment mechanism we described; in particular a @Comment does not need to be a valid entry, i.e. it can for example skip comas between two fields
http://maverick.inria.fr/~Xavier.Decoret/resources/xdkbibtex/bibtex_summary.html#comment

In my case "@Comment" are always on one line and no line is very long.
So I just fixed it with a "fgets" instead of "fread" and a small exeption in the parseFile function : 
```
public function parseFile($file)
{
    $handle = fopen($file, 'r');
    try {
        $this->reset();
        while (!feof($handle)) {
            $buffer = fgets($handle);//fread($handle, 128);
            if (strpos($buffer, '@Comment{') === false) {
                $this->parse($buffer);
            }
        }
        $this->throwExceptionIfReadingEntry("\0");
    } finally {
        fclose($handle);
    }
}

```

"@Comment" in my .bib file : 
```
@Comment{jabref-meta: databaseType:bibtex;}
@Comment{jabref-entrytype: Poster: req[author;title;year;organization] opt[abstract]}
```
I found that they are automaticaly added by JabRef (http://www.jabref.org/) at the end of the .bib file.
https://github.com/JabRef/jabref/issues/2944

Have a nice day!

**Comment by renanbr** [comment]
Thanks for reporting this issue.

Your solution is a good start, and I would like to fix it for all the cases I found in the link you bring up:

```bibtex
@Comment{jabref-meta: databaseType:bibtex;}

@Comment{jabref-entrytype: Poster: req[author;title;year;organization] opt[abstract]}

@Comment{
  @Book{steward03,
    author =	 {Martha Steward},
    title =	 {Cooking behind bars},
    publisher =	 {Culinary Expert Series},
    year =	 2003
  }
}
```

If I got your point, in the exemple above, the `@Book` entry should be take in account, the rest ignored, and no error thrown.

A alternative way to solve the problem is skiping/ignoring `@comment` type:
1. **Not** trigger the listeners when a `@comment` type is found; and
2. Then switch the current state to `COMMENT` at the moment this "type" is found

Advantage: We can keep the parser able to read char-by-char, as currently it does.

What do you think?

Note: This change could break this test: https://github.com/renanbr/bibtex-parser/blob/master/tests/Parser/CommentTest.php#L34-L67

**Comment by renanbr** [comment]
JabRef case is fixed (not yet released):
- [fix](https://github.com/renanbr/bibtex-parser/commit/e0a0dd62a4bd5054e69071b201e2638243baa0cf)
- [test](https://github.com/renanbr/bibtex-parser/commit/5ba071451d7437e3059db484c0e4089b8d870989)
- [build result](https://travis-ci.org/renanbr/bibtex-parser/builds/377531234)

---

I'm having problems in understanding this sentence:

> Actually, the rule is that everything from the @comment and to the end of line is ignored.

So, what's the braces' role for an comment "entry"?

In the bibtex bellow, I'm willing to consider the existence of 3 entries (note the missing "}" at the end of the file):

```bibtex
@comment{@book{title={I do exist}}}

@comment{
  @book{title={I really do exist}}
}

@comment{ bla bla bla
@book{title={I'm real}}
```

A first approach is available : https://github.com/renanbr/bibtex-parser/tree/comment-is-not-an-entry

**Comment by renanbr** [comment]
@Nahihilo I pushed another branch with a better and simpler fix IMO, see #51

I'm going to merge it if you don't stand up against it
